tactics/abstract.ml

```diff
@@ -100,3 +100,6 @@ let abstract_subproof ~opaque tac =
 
 let tclABSTRACT ?(opaque=true) name_op tac =
   abstract_subproof ~opaque ~name_op tac
+
+let () = Hook.set Proof.abstract_hook
+    (fun tac -> tclABSTRACT None tac)
```

Comment:
This seems to indicate that the right place for abstract is in `proofs`? WDYT?

Reply:
No, this is the wrong way around. Abstract should live in the proof engine. This PR is relying on pieces of code that should disappear.

Reply:
Moving it to proofs is making it closer to `engine`, I'm not sure I understand what you mean?

Reply:
I'll see if I can do a cleanup, but in another PR

Reply:
It's not trivial to push abstract below tactics as it uses some stuff from there 'tclCOMPLETE, intros)